src/cascadia/TerminalSettingsModel/ProfileEntry.cpp

```diff
@@ -0,0 +1,41 @@
+#include "pch.h"
+#include "ProfileEntry.h"
+#include "JsonUtils.h"
+
+#include "ProfileEntry.g.cpp"
+
+using namespace Microsoft::Terminal::Settings::Model;
+using namespace winrt::Microsoft::Terminal::Settings::Model::implementation;
+
+static constexpr std::string_view ProfileKey{ "profile" };
+
+ProfileEntry::ProfileEntry() noexcept :
+    ProfileEntry{ winrt::hstring{} }
+{
+}
+
+ProfileEntry::ProfileEntry(const winrt::hstring& profile) noexcept :
+    ProfileEntryT<ProfileEntry, NewTabMenuEntry>(NewTabMenuEntryType::Profile),
+    _ProfileName{ profile }
+{
+}
+
+Json::Value ProfileEntry::ToJson() const
+{
+    auto json = NewTabMenuEntry::ToJson();
+
+    // We always store the GUID of the profile since that is less
+    // errorprone in the long term.
+    JsonUtils::SetValueForKey(json, ProfileKey, _Profile.Guid());

```

Comment:
Does this mean that the user can input the profile name into the json, but at a later point when the settings file gets rewritten (say from hitting 'save' in the SUI), the profile name gets replaced with the profile guid? I wonder if we should only accept profile guids and only write profile guids, instead of accept either guid or name but only write guids.

Reply:
Yes, that is correct. I think it is fair to do it this way; by accepting/reading both names and GUIDs it's much easier for users to write this config while there is no UI yet.